# Fix a lint error in http_wait_response's call to parse_response
# The error shows up in studio but not gcc
# Developed in-house
# Plan to send this upstream.
diff -rupN old/dirmngr/http.c new/dirmngr/http.c
--- old/dirmngr/http.c	2018-04-12 21:48:54.231537397 +0000
+++ new/dirmngr/http.c	2018-04-12 23:16:03.497480859 +0000
@@ -157,7 +157,7 @@ static gpg_error_t send_request (http_t
 				 const char *srvtag, unsigned int timeout,
                                  strlist_t headers);
 static char *build_rel_path (parsed_uri_t uri);
-static gpg_error_t parse_response (http_t hd);
+static gpg_err_code_t parse_response (http_t hd);
 
 static gpg_error_t connect_server (const char *server, unsigned short port,
                                    unsigned int flags, const char *srvtag,
@@ -1067,6 +1067,7 @@ gpg_error_t
 http_wait_response (http_t hd)
 {
   gpg_error_t err;
+  gpg_err_code_t rc;
   cookie_t cookie;
   int use_tls;
 
@@ -1112,10 +1113,12 @@ http_wait_response (http_t hd)
       return err;
     }
 
-  err = parse_response (hd);
+  rc = parse_response (hd);
 
-  if (!err)
+  if (!rc)
     err = es_onclose (hd->fp_read, 1, fp_onclose_notification, hd);
+  else
+    err = gpg_err_make (default_errsource, rc);
 
   return err;
 }
